docs: update various documentation for ECL 4 - FRONT-4234 (#3224)
ec-europa/europa-component-library: docs/Migrating-v4.md

# ECL v4 migration notes

> The following guidelines aim to facilitate migration between ECL v3 and v4.

Reviewer: ```suggestion
The following guidelines aim to facilitate the migration from ECL v3 to v4.
```

- [Js modifications](#js-modifications)
- [Style modifications](#style-modifications)
- [Component modifications](#component-modifications)
- [Customizing the display](#customizing-the-display)

## Js modifications

- The ECL javascript is not bundling anymore `pikaday`, therefore this library needs to be loaded by the project using ECL, it is only
  needed when a `datepicker` instance is present in the page.
- There is no hard dependency on `moment.js` anymore, this library is only needed when customising the default format for the dates, the default in ECL is `DD-MM-YYYY`. If the format is customised but moment is not loaded there will not be errors in the console but the chosen format would not be consistently shown.

> - Callback are now provided for some components, and can be used to trigger custom code when needed

Reviewer: ```suggestion
- A simple event manager is now available in ECL, some of the components are defining and emitting events that the user can subscribe to and execute a callback when they are emitted. The list of these events and component supporting them will increase in the future.
```

#### Description list

> - Lists with links, in most cases, expect the `ecl-link--standalone` class to prevent the underlining. The only excpetion is the taxonomy list.

Reviewer: ```suggestion
- Lists with links, in most cases, expect the `ecl-link--standalone` class to prevent the underlining. The only exception is the taxonomy list.
```

- Tags now wrap on several lines by default. New twig parameter `nowrap` added to force display of the tag on one line.
- Added possibilty to handle set of tags. Twig template and corresponding css have been added.

## Customizing the display

ECL4 introduces new ways to customize the look and feel of elements, by changing styles and component display

### Using CSS variables

> The easiest way to change the look and feel of ECL is by using the provided CSS variables. They are provided for most common styles values.

Reviewer: can we use custom properties instead of variables everywhere we relate to an --variable ?  Variables can be confusing, while custom properties are specifically identifying what we mean.

- typgraphy
- shadow

By default, these variables take values from the SASS variables. You can freely override any variable you want in a custom CSS file loaded after ECL one.

Example:

```
your_custom_css_file.css

:root {
  --ecl-color-primary: #00f;
  --ecl-color-primary-120: #00d;
  ...
  --ecl-font-m: normal normal 400 1rem/1.75rem "Comic Sans ms",arial,sans-serif;
  ...
  --ecl-shadow-1: 1px 1px 2px rgba(0, 0, 0, 0.5);
  ...
}
```

Note: we are using shorter aliases internally to limit the size of the compiled css file, but it is recommended to use the full variable name if you want to override them

### Using SASS variables

If you are building ECL directly on your side, you can change the values of the SASS variables before compilation.
This is not the recommended way to customize ECL, as it may conflict with future updates (CSS variables are prefered), but this is an option if you have full control over the ECL implementation.

> SASS variables are defined in the files `src/themes/[your_theme]/variables/*.scss`. There are two theme available by default (EC and EU).

Reviewer: ```suggestion
SASS variables are defined in the files `src/themes/[your_theme]/variables/*.scss`. There are two themes available by default (EC and EU).
```

> SASS variables are defined in the files `src/themes/[your_theme]/variables/*.scss`. There are two theme available by default (EC and EU).

Reviewer: we do refer to the repository filesystem, wouldn't it better to refer to the packages content, instead?

> You can edit the values, and recompile ECL css by running the command `yarn dist` at the root of the project. The compiled files are available in `/dist/packages/[your_theme]`

Reviewer: i
```suggestion
You can edit the values, and recompile ECL css by running the command `yarn` at the root of the project. The compiled files are available in `/src/presets/[ec/eu]/dist`.
```
I think that suggesting to run yarn dist only to get the resources compiled is an overkill, running yarn is enough to get them compiled and they can be retrieved from the presets folders.
